src/main/java/org/opensearch/neuralsearch/processor/combination/AbstractScoreCombinationTechnique.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor.combination;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Base class for score normalization technique
+ */
+public abstract class AbstractScoreCombinationTechnique {
```

Comment:
abstract class makes sense as we going to add more combination techniques soon and big part of the implementation can be shared

Reply:
How about using interface for some flexibility in the future especially when it won't have any instance variables?